### DISCUSSION TITLE
Why does ccache think -werror is compiler_only_args?

### DISCUSSION THREAD

**Discussion by zcfh** [ccache/ccache#1122] [Support]
Here's an example.
a.cc
``` c++
#include "stdio.h"

#define print(...) printf(__VA_ARGS__)

int main() {
    char a = "hello";
    print("1: %s, \n
           2: %s\n",
           a, a);
}

```
And preprocessing will issue a warning. 
```
tutorial/a.cc:7:11: warning: missing terminating " character
     print("1: %s, \n
           ^
tutorial/a.cc:8:19: warning: missing terminating " character
            2: %s\n",
                   ^
```
Preprocessed output
```
....
#define print(...) printf(__VA_ARGS__)


# 5 "tutorial/a.cc"
int main() {
    char a = "hello";
    printf("1: %s, \n 2: %s\n", a, a)

                ;
}
```
Because of the preprocessed file, multiple lines of characters are merged into one line, and no warning occurs when you compile again. This causes ccache to compile successfully, but not directly.

**Comment by jrosdahl** [comment]
Sorry, but I don't quite understand what you mean. What is it that doesn't work like you expect? Please provide more details.

That said, I guess (but please don't have me guess) that you are using `run_second_cpp = false` in the configuration (or set `CCACHE_NOCPP2=1`). That will alter how the compiler behaves in some situations, and this is one. If that's your issue, I think you'll either have to _not_ use `run_second_cpp = false` or live with the fact that you get different results in some cases. That's the trade-off with `run_second_cpp`.

For background why `-Werror` isn't sent to the preprocessor, see 717fcb1ec952ceda871bb008128a0a834fdc70f3.
